docs/Deployment-with-dokku.adoc

```diff
@@ -0,0 +1,187 @@
+```
+
+For Postgres:
+
+```bash
+dokku postgres:create hello-world
+dokku postgres:link hello-world hello-world
+```
+
+For RabbitMQ:
+
+```bash
+dokku rabbitmq:create hello-world
+dokku rabbitmq:link hello-world hello-world
+```
+
+> Note: It appears the rabbitmq create hangs sometimes and doesn't do anything. Just press `ctrl` + `c` to confirm and it will continue just fine.
+
+To confirm you can type `dokku config hello-world` and should see all the previously mentioned services are linked there.
+
+---
+
+We need to setup some volumes for persistent storage
+
+```bash
+sudo -u dokku mkdir -p /home/dokku/hello-world/uploads
+sudo -u dokku mkdir -p /home/dokku/hello-world/backups
+dokku docker-options:add hello-world deploy,run "-v /home/dokku/hello-world/uploads:/APP/uploads"
+dokku docker-options:add hello-world deploy,run "-v /home/dokku/hello-world/backups:/APP/backups"
+```
+
+=== Set Environment Configuration
+
+Running beavy on the server means we need to expose the proper environment variables
+
+`dokku config:set hello-world DOKKU_DOCKERFILE_PORT=5000 BEAVY_ENV=PRODUCTION BEAVY_CONFIG_FROM_ENV=1 DOKKU_DOCKERFILE_START_CMD='/app/run.sh' C_FORCE_ROOT=1`
+
+This informs dokku that we will be hosting the project on port 5000, tells beavy to run in the production mode and understand the environment variables as configuration option. This way it will automatically detect the services we linked perviously. In this mode, you can set any (first-level) configuration option of beavy via `dokku config:set`, too.
+
+In details this is what they do:
+
+ - `DOKKU_DOCKERFILE_PORT=5000` tells dokku that we will be serving on port `5000`.
+ - `BEAVY_ENV=PRODUCTION` sets beavy into production mode.
+ - `BEAVY_CONFIG_FROM_ENV=1` enabled Environment-variables overwrite of configuration values. This is necessary because dokku exposes connections to the backend services via environment variables we need to configure the service
+ - `DOKKU_DOCKERFILE_START_CMD='/app/run.sh'` informs dokku to run a custom script we are shipping in the container, which allows us to start either web or worker processes and scale properly with dokku features
+  - `C_FORCE_ROOT=1` informs celerey (our worker system) that is fine to run as root. At the moment, dokku makes somes troubles if we run the service as a different user. But constraint in the container that doesn't matter that much.
+
+== Your first push
+
+Now you can try to push the app for the first time, to see if it builds properly. If you haven't yet created a local `config.yml` for your app, do that and force add it to git (`git add -f config.yml`) – otherwise it will break (with `lstat config.yml: no such file or directory`).
+
+=== Assets
+
+In the build, the system currently assumes the pre-built assets are shipped in the repository. Once we have travis set up, that will happen automatically, but if you want to push from your local repository, the build will fail because it can't find them.
```

Comment:
No, this is meant to be like this. I added this here so you can have a first check whether the deploying works before having to push to travis.

But taking the considerations in place and that this is a long todo list anyways, it might make sense to drop this or have it only in a "troubleshoot" section below....